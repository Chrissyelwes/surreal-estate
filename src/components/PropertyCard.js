import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { solid } from "@fortawesome/fontawesome-svg-core/import.macro";
import PropTypes from "prop-types";

const PropertyCard = (props) => {
  const { bathrooms, bedrooms, city, price, title, type } = props;

  return (
    <div className="property-card" data-testid="property-card">
      <div className="property-card_items">
        <div className="property-card_title">{title}</div>
        <div className="property-card_type-city">
          <FontAwesomeIcon
            className="property-card_icon-house"
            data-testid="icon_house"
            icon={solid("house")}
          />
          {type}
          -
          <FontAwesomeIcon
            className="property-card_icon-building"
            data-testid="icon_building"
            icon={solid("building")}
          />
          {city}
        </div>
        <div className="property-card_bathrooms">
          <FontAwesomeIcon
            className="property-card_icon-toilet"
            data-testid="icon_toilet"
            icon={solid("toilet")}
          />
          {bathrooms}
        </div>
        <div className="property-card_bedrooms">
          <FontAwesomeIcon
            className="property-card_icon-bed"
            data-testid="icon_bed"
            icon={solid("bed")}
          />
          {bedrooms}
        </div>
        <div className="property-card_price">Â£{price}</div>
        <p>
          <a href="mailto:someone@example.com">EMAIL</a>
          <FontAwesomeIcon icon="fa-solid fa-building" />
        </p>
      </div>
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
};

export default PropertyCard;
