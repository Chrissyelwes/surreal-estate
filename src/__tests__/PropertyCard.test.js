import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "stub title",
    type: "flat",
    city: "Manchester",
    bathrooms: 16,
    bedrooms: 18,
    price: 100,
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        city={validProps.city}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        city={validProps.city}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
      />
    );

    expect(validProps.title).toBe("stub title");
    expect(validProps.type).toBe("flat");
    expect(validProps.city).toBe("Manchester");
    expect(validProps.bathrooms).toBe(16);
    expect(validProps.bedrooms).toBe(18);
    expect(validProps.price).toBe(100);
  });

  it("renders correct class for props", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        city={validProps.city}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
      />
    );

    expect(getByText("stub title")).toHaveClass("property-card_title");
    expect(getByText("flat")).toHaveClass("type");
    expect(getByText("Manchester")).toHaveClass("city");
    expect(getByText(16)).toHaveClass("bathrooms");
    expect(getByText(18)).toHaveClass("bedrooms");
    expect(getByText(100)).toHaveClass("property-card_price");
  });
});
