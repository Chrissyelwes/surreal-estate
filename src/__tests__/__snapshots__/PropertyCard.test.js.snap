// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card_wrapper"
    data-testid="property-card_wrapper"
  >
    <div
      class="property-card_items"
    >
      <div
        class="property-card_title"
      >
        stub title
      </div>
      <div
        class="property-card_type-city"
      >
        <div
          class="type"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-house property-card_icon-house"
            color="#BD9561"
            data-icon="house"
            data-prefix="fas"
            data-testid="icon_house"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
              fill="currentColor"
            />
          </svg>
          flat
        </div>
        |
        <div
          class="city"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-building property-card_icon-building"
            color="#BD9561"
            data-icon="building"
            data-prefix="fas"
            data-testid="icon_building"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 0C21.5 0 0 21.5 0 48V464c0 26.5 21.5 48 48 48h96V432c0-26.5 21.5-48 48-48s48 21.5 48 48v80h96c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48H48zM64 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V240zm112-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zm80 16c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V240zM80 96h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V112c0-8.8 7.2-16 16-16zm80 16c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V112zM272 96h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V112c0-8.8 7.2-16 16-16z"
              fill="currentColor"
            />
          </svg>
          Manchester
        </div>
      </div>
      <div
        class="property-card_bathrooms-bedrooms"
      >
        <div
          class="bathrooms"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-toilet property-card_icon-toilet"
            color="#BD9561"
            data-icon="toilet"
            data-prefix="fas"
            data-testid="icon_toilet"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 0C10.7 0 0 10.7 0 24S10.7 48 24 48h8V196.9c-1.9 1.4-3.8 2.9-5.6 4.4C10.9 214.5 0 232.9 0 256c0 46.9 14.3 84.1 37 112.5c14.2 17.7 31.1 31.3 48.5 41.8L65.6 469.9c-3.3 9.8-1.6 20.5 4.4 28.8s15.7 13.3 26 13.3H352c10.3 0 19.9-4.9 26-13.3s7.7-19.1 4.4-28.8l-19.8-59.5c17.4-10.5 34.3-24.1 48.5-41.8c22.7-28.4 37-65.5 37-112.5c0-23.1-10.9-41.5-26.4-54.6c-1.8-1.5-3.7-3-5.6-4.4V48h8c13.3 0 24-10.7 24-24s-10.7-24-24-24H24zM384 256.3c0 1-.3 2.6-3.8 5.6c-4.8 4.1-14 9-29.3 13.4C320.5 284 276.1 288 224 288s-96.5-4-126.9-12.8c-15.3-4.4-24.5-9.3-29.3-13.4c-3.5-3-3.8-4.6-3.8-5.6l0-.3 0-.1c0-1 0-2.5 3.8-5.8c4.8-4.1 14-9 29.3-13.4C127.5 228 171.9 224 224 224s96.5 4 126.9 12.8c15.3 4.4 24.5 9.3 29.3 13.4c3.8 3.2 3.8 4.8 3.8 5.8l0 .1 0 .3zM328.2 384l-.2 .5 0-.5h.2zM112 64h32c8.8 0 16 7.2 16 16s-7.2 16-16 16H112c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
              fill="currentColor"
            />
          </svg>
          16
        </div>
        |
        <div
          class="bedrooms"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bed property-card_icon-bed"
            color="#BD9561"
            data-icon="bed"
            data-prefix="fas"
            data-testid="icon_bed"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 32c17.7 0 32 14.3 32 32V320H288V160c0-17.7 14.3-32 32-32H544c53 0 96 43 96 96V448c0 17.7-14.3 32-32 32s-32-14.3-32-32V416H352 320 64v32c0 17.7-14.3 32-32 32s-32-14.3-32-32V64C0 46.3 14.3 32 32 32zM176 288c-44.2 0-80-35.8-80-80s35.8-80 80-80s80 35.8 80 80s-35.8 80-80 80z"
              fill="currentColor"
            />
          </svg>
          18
        </div>
      </div>
      <div
        class="property-card_price"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sterling-sign property-card_icon-price"
          color="#BD9561"
          data-icon="sterling-sign"
          data-prefix="fas"
          data-testid="icon_price"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M144 160.4c0-35.5 28.8-64.4 64.4-64.4c6.9 0 13.8 1.1 20.4 3.3l81.2 27.1c16.8 5.6 34.9-3.5 40.5-20.2s-3.5-34.9-20.2-40.5L249 38.6c-13.1-4.4-26.8-6.6-40.6-6.6C137.5 32 80 89.5 80 160.4V224H64c-17.7 0-32 14.3-32 32s14.3 32 32 32H80v44.5c0 17.4-4.7 34.5-13.7 49.4L36.6 431.5c-5.9 9.9-6.1 22.2-.4 32.2S52.5 480 64 480H320c17.7 0 32-14.3 32-32s-14.3-32-32-32H120.5l.7-1.1C136.1 390 144 361.5 144 332.5V288H256c17.7 0 32-14.3 32-32s-14.3-32-32-32H144V160.4z"
            fill="currentColor"
          />
        </svg>
        100
      </div>
      <button
        class="button_type-3"
        type="button"
      >
        Email
      </button>
    </div>
  </div>
</DocumentFragment>
`;
